### ISSUE TITLE
Fulfillments and Errors are Responses, Not Requests

### ISSUE THREAD

**Issue by dappelt** [emschwartz/ilp3#14]
The design principal [Fulfillments and Errors are Responses, Not Requests](https://github.com/emschwartz/ilp3#1-fulfillments-and-errors-are-responses-not-requests) in combination with [Recommend Using HTTP](https://github.com/emschwartz/ilp3#3-recommend-using-http) limits payment timeouts in ilp3 to that of HTTP timeouts. In consequence, only very fast ledgers will be compatible with ilp3.  Also, if a single TCP connection along the path drops the payment fails and the connector that had his connection dropped potentially losing funds. This seems to be very fragile.

**Comment by emschwartz** [comment]
> limits payment timeouts in ilp3 to that of HTTP timeouts. In consequence, only very fast ledgers will be compatible with ilp3

That is intentional. As mentioned [here](https://github.com/emschwartz/ilp3#4-fast-htlas-only), I think we should only focus on fast HTLAs (which means no escrow until ledgers are super fast).

> Also, if a single TCP connection along the path drops the payment fails and the connector that had his connection dropped potentially losing funds. This seems to be very fragile.

There are two ways I would look at this: first, think of payments as being really small, small enough that it doesn't matter if one gets lost. In such a system I think you'd worry much more about achieving maximum speed rather than accounting for every single cent.

If that isn't good enough, you could also add a transfer ID to the ledger layer protocol and make transfers idempotent within a short window of time. That way, if your connection drops and you don't get the fulfillment back you could just retry the transfer (quickly, because as you pointed out the expiry window would be very short) and you'd ideally get the fulfillment then.

**Comment by adrianhopebailie** [comment]
> That is intentional. As mentioned here, I think we should only focus on fast HTLAs (which means no escrow until ledgers are super fast).

I agree with this but think it is also use case specific. It's completely acceptable for a retail payment to take up to 30 seconds to complete for example.

So your definition of "fast" may be different to mine.

**Comment by emschwartz** [comment]
If you need 1000 chunks to complete your retail payment I don't think you'd
be too happy about each chunk taking multiple seconds or longer ;)
